plugin/src/main/java/com/denizenscript/denizen/events/block/BlockEquipsItemScriptEvent.java

```diff
@@ -0,0 +1,104 @@
+package com.denizenscript.denizen.events.block;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockDispenseArmorEvent;
+
+public class BlockEquipsItemScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // block equips <item>
+    //
+    // @Group Block
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a block dispenses armor and is equipped to an entity.
+    //
+    // @Switch on:<entity> to only process the event if the entity having the armor equipped matches the entity input.
+    //
+    // @Context
+    // <context.item> returns the ItemTag of the armor being dispensed.
+    // <context.entity> returns the EntityTag of the entity having the armor equipped.
+    // <context.location> returns the LocationTag of the dispenser.
+    //
+    // @Player when the equipped entity is a player.
+    //
+    // @NPC when the equipped entity is an NPC.
+    //
+    // @Determine
+    // ItemTag to set the armor being equipped.
+    //
+    // @Warning Determined armor types must match or armor will be assigned incorrect slots (for example, if the original item was a helmet but the new item is boots, the boots will be assigned to the helmet slot and will not display properly). Determining a non-armor item will be dispensed normally.
+    //
+    // @Example
+    // # Will cause leather armor to be dispensed like a normal item and not be equipped on an armor stand.
+    // on block equips leather* on:armor_stand:
+    // - determined cancelled
+    //
+    // @Example
+    // # Will equip a golden helmet if a leather helmet is originally being equipped.
+    // on block equips leather_helmet:
+    // - determine golden_helmet
+    // -->
+
+    public BlockEquipsItemScriptEvent() {
+        registerCouldMatcher("block equips <item>");
+        registerSwitches("on");
+        this.<BlockEquipsItemScriptEvent, ItemTag>registerOptionalDetermination(null, ItemTag.class, (evt, context, item) -> {
+            evt.event.setItem(item.getItemStack());
+            return true;
+        });

```

Comment:
Modern/new determinations should usually have a prefix, prefix-less determination support is for old events being updated.
Also shouldn't need `registerOptionalDetermination` here, as any item is valid - can just `registerDetermination`.